src/plugins/intel_gpu/tests/test_cases/concatenation_gpu_test.cpp

```diff
@@ -1285,4 +1285,154 @@ INSTANTIATE_TEST_SUITE_P(smoke,
                             TestParamType_concat(1, { 8, 32 }, 2, 2, false)
                         ),
                         concat_gpu::PrintToStringParamName);
+
+template <typename Type>
+struct concat_gpu_4d_explict : public concat_gpu {
+public:
+    cldnn::memory::ptr run_concat_network(std::vector<std::vector<std::vector<std::vector<std::vector<Type>>>>> input, format::type fmt, ExecutionConfig config) {
+        auto data_type = type_to_data_type<Type>::value;
+        auto& engine = get_test_engine();
+        const size_t batch_num = testing::get<0>(GetParam());
+        const std::vector<size_t> in_features = testing::get<1>(GetParam()); // only use first element.
+        const size_t input_y = testing::get<2>(GetParam());
+        const size_t input_x = testing::get<3>(GetParam());
+        size_t output_f = in_features[0];
+
+        topology topology;
+
+        std::vector<memory::ptr> in_memory;
+        std::vector<primitive_id> input_ids;
+
+        // input0 -- eltwise -- conv -- concat -- output
+        //         /         \        /
+        // input1 -           --------
```

Comment:
does this test fail without your fix? I think this is a bit different from our target pattern. To my understanding, our target pattern is 1) inputs of concat is cldnn 2) one of input has another user which is running on onednn.
However, in this topology, input of concat is actually cldnn. Then this will just use explicit concat even without your fix.

Reply:
Unit test topology has changed.